### PR TITLE
Browser fixes

### PR DESCRIPTION
**Pull request by bretambrose**
A grab bag of browser shim fixes:

* Update http sample to use new http interface
* Webpack error in http module
* Corked event args not unpacking properly leading to undefined arguments
* mqtt connect timeout default should be a reasonable value in milliseconds, not seconds
* activate http streams in browser http sample
* rework HttpClientConnection constructor to not be confusing and erroneously ambiguous with null/undefined parameters
* audit fix for vulnerable dependencies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/browser/http.ts [hunk 1]
```diff
@@ -356,7 +349,6 @@ export class HttpClientStream extends BufferedEventEmitter {
         }
         this.emit('data', data);
         this.emit('end');
-        this.connection.close();
```
Comment (bretambrose) [inline lib/browser/http.ts:359]: Only change I'm not certain about.  Neither was Justin.

File: lib/browser/http.ts [hunk 2]
```diff
@@ -169,29 +169,22 @@ export class HttpClientConnection extends BufferedEventEmitter {
...
-    constructor(host_name: string, port: number, socket_options?: SocketOptions, tls_options?: TlsConnectionOptions, proxy_options?: HttpProxyOptions);
-    constructor(bootstrap: ClientBootstrap, host_name: string, port: number, socket_options?: SocketOptions, tls_options?: TlsConnectionOptions, proxy_options?: HttpProxyOptions);
     constructor(
-        bootstrapOrHost: ClientBootstrap | string,
-        hostOrPort: string | number,
-        portOrSocketOptions?: number | SocketOptions,
-        socketOptionsOrTlsOptions?: SocketOptions | TlsConnectionOptions,
-        tlsOptionsOrProxyOptions?: TlsConnectionOptions | HttpProxyOptions,
-        maybeProxyOptions?: HttpProxyOptions,
+        bootstrap: ClientBootstrap,
```
Comment (justinboswell) [inline lib/browser/http.ts:173]: Any reason not to make this ```| undefined```?

File: lib/common/event.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ type EventKey = string | symbol;
 class BufferedEvent {
     public next?: BufferedEvent;
     public args: any[];
-    constructor(public event: EventKey, ...args: any[]) {
+    constructor(public event: EventKey, args: any[]) {
```
Comment (justinboswell) [inline lib/common/event.ts:22]: OH NO I'M SO SORRY

File: lib/browser/http.ts (hunk 2 above)
Comment (bretambrose) [reply lib/browser/http.ts:173]: Seems like it would be ok, will give it a try

**Pull request merged**